test/test-poll-oob.c

```diff
@@ -0,0 +1,203 @@
+    return;
+
+  uv_poll_stop(&poll_req[0]);
+  uv_poll_stop(&poll_req[1]);
+  uv_close((uv_handle_t*) &server_handle, NULL);
+  uv_close((uv_handle_t*) &client_handle, NULL);
+  uv_close((uv_handle_t*) &peer_handle, NULL);
+  uv_close((uv_handle_t*) idle, NULL);
+}
+
+static void poll_cb(uv_poll_t* handle, int status, int events) {
+  char buffer[5];
+  int n;
+  int fd;
+
+  ASSERT(0 == uv_fileno((uv_handle_t*)handle, &fd));
+  memset(buffer, 0, 5);
+
+  if (events & UV_PRIORITIZED) {
+    do
+      n = recv(client_fd, &buffer, 5, MSG_OOB);
+    while (n == -1 && errno == EINTR);
+    ASSERT(errno != EINVAL);
+    ASSERT(n > 0);
+    cli_pr_check = 1;
+  }
+  if (events & UV_READABLE) {
+    if (fd == client_fd) {
+      n = recv(client_fd, &buffer, 5, 0);
+      ASSERT(errno != EINVAL);
+      if (errno == EINTR) {
+        return;
+      }
+      if (cli_rd_check == 1) {
+        ASSERT(strncmp(buffer, "world", n) == 0);
+        ASSERT(5 == n);
+        cli_rd_check = 2;
+      }
+      if (cli_rd_check == 0) {
+        ASSERT(n == 4);
+        ASSERT(strncmp(buffer, "hello", n) == 0);
+        cli_rd_check = 1;
+        do {
+          n = recv(client_fd, &buffer, 5, 0);
+          ASSERT(n >= 0 || errno == EINTR)
+          if (n > 0) {
+            ASSERT(n == 5);
+            ASSERT(strncmp(buffer, "world", n) == 0);
+            cli_rd_check = 2;
+          }
+        } while (n > 0);
+
+        ASSERT(got_eagain());
+      }
+    }
+    if (fd == server_fd) {
+      n = recv(server_fd, &buffer, 3, 0);
+      ASSERT(errno != EINVAL);
+      if (errno == EINTR) {
+        return;
+      }
+      ASSERT(3 == n);
+      ASSERT(strncmp(buffer, "foo", n) == 0);
+      srv_rd_check = 1;
+      uv_poll_stop(&poll_req[1]);
+    }
+  }
+  if (events & UV_WRITABLE) {
+    do {
+      n = send(client_fd, "foo", 3, 0);
+    } while (n < 0 && errno == EINTR);
+    ASSERT(3 == n);
+  }
+}
+
+static void connection_cb(uv_stream_t* handle, int status) {
+  int r;
+
+  ASSERT(0 == status);
+  ASSERT(0 == uv_accept(handle, (uv_stream_t*) &peer_handle));
+  ASSERT(0 == uv_fileno((uv_handle_t*) &peer_handle, &server_fd));
+  ASSERT(0 == uv_poll_init_socket(uv_default_loop(), &poll_req[0], client_fd));
+  ASSERT(0 == uv_poll_init_socket(uv_default_loop(), &poll_req[1], server_fd));
+  ASSERT(0 == uv_poll_start(&poll_req[0],
+                            UV_PRIORITIZED | UV_READABLE | UV_WRITABLE,
+                            poll_cb));
+  ASSERT(0 == uv_poll_start(&poll_req[1],
+                            UV_READABLE,
+                            poll_cb));
+  do {
+    r = send(server_fd, "hello", 5, MSG_OOB);
+  } while (r < 0 && errno == EINTR);
+  ASSERT(5 == r);
+
+  do {
+    r = send(server_fd, "world", 5, 0);
+  } while (r < 0 && errno == EINTR);
+  ASSERT(5 == r);
+
+  ASSERT(0 == uv_idle_start(&idle, idle_cb));
+}
+
+
+TEST_IMPL(poll_oob) {
+  struct sockaddr_in addr;
+  int addrlen = sizeof(addr), r = 0;
+  uv_loop_t* loop;
```

Comment:
Separate declaration from assignment. I think `addrlen` variable is superfluous.